controller/lifecycle/lifecycle.go

```diff
@@ -158,6 +198,45 @@ func (l *LifecycleManager) Reconcile(ctx context.Context, req ctrl.Request, inst
 	return result, nil
 }
 
+func (l *LifecycleManager) updateStatus(ctx context.Context, original runtime.Object, current RuntimeObject, log *logger.Logger, sentryTags sentry.Tags) error {
+
+	currentUn, err := runtime.DefaultUnstructuredConverter.ToUnstructured(current)
+	if err != nil {
+		return err
+	}
+
+	originalUn, err := runtime.DefaultUnstructuredConverter.ToUnstructured(original)
+	if err != nil {
+		return err
+	}
+
+	currentStatus, hasField, err := unstructured.NestedFieldCopy(currentUn, "status")
+	if !hasField || err != nil {
+		return err
+	}
+	originalStatus, hasField, err := unstructured.NestedFieldCopy(originalUn, "status")
+	if !hasField || err != nil {
+		return err
+	}
+
+	if equality.Semantic.DeepEqual(currentStatus, originalStatus) {
+		log.Info().Msg("skipping status update, since they are equal")
+		return nil
+	}
+
+	log.Info().Msg("updating resource status")
+	err = l.client.Status().Update(ctx, current)
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			sentry.CaptureError(err, sentryTags, sentry.Extras{"message": "Updating of instance status failed"})
```

Comment:
🟡  i think this is a special case, and we might not want to sentry log it. 
As soon as we have event processing on the resource, this error will happen, and we cannot do anything about it.

So either we even make it part of the lifecycle api, if it should capture here, or we do not send it ❓
The log statement is fine, but maybe on the Warn level

Reply:
i missed the `!` sign when reviewing, so actually its fine to capture the other kind of errors